be/src/olap/rowset/segment_v2/inverted_index/query/conjunction_query.h

```diff
@@ -0,0 +1,70 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "CLucene/index/IndexReader.h"
+#include "CLucene/index/IndexVersion.h"
+#include "CLucene/index/Term.h"
+#include "CLucene/search/query/TermIterator.h"
+#include "roaring/roaring.hh"
+
+CL_NS_USE(index)
+
+namespace doris {
+
+class ConjunctionQuery {
+public:
+    ConjunctionQuery(IndexReader* reader);
+    ~ConjunctionQuery();
+
+    void set_conjunction_ratio(int32_t conjunction_ratio) {
+        _conjunction_ratio = conjunction_ratio;
+    }
+
+    void add(const std::wstring& fieldName, const std::vector<std::wstring>& wterms);
+    void search(roaring::Roaring& roaring);
+
+private:
+    void search_by_bitmap(roaring::Roaring& roaring);
+
+    void search_by_skiplist(roaring::Roaring& roaring) {
+        int32_t doc = 0;
+        while ((doc = next_doc()) != INT32_MAX) {
+            roaring.add(doc);
+        }
+    }
+
+    int32_t next_doc() { return do_next(_lead1.nextDoc()); }
```

Comment:
next_doc is only used once and is simple, is it necessary to create a new function for it?